primitives/bytes/b4.go

```diff
@@ -108,3 +108,48 @@ func FromUint32(v uint32) B4 {
 func (h B4) ToUint32() uint32 {
 	return binary.LittleEndian.Uint32(h[:])
 }
+
+/* -------------------------------------------------------------------------- */
+/*                                Comparable                                  */
+/* -------------------------------------------------------------------------- */
+
+// NOTE: IsLessThan and IsGreaterThan are not implemented to avoid opinionating a
+// canonical sorting algorithm for B4, the 4-byte vector. A standard cmp function
+// would require IsLessThan and IsGreaterThan and behave something like this:
+//
+//	func cmp(a, b B4) int {
+//		if a.IsLessThan(b) {
+//			return -1
+//		}
+//		if a.IsGreaterThan(b) {
+//			return 1
+//		}
+//		return 0
+//	}
+
+// Equals returns true if h and b are equal.
+func (h B4) Equals(b B4) bool {
+	return h == b
+}
+
+// IsLessThanOrEqualTo returns true if h is less than or equal to b. This compares bytes from
+// most significant to least significant in little-endian order.
+func (h B4) IsLessThanOrEqualTo(b B4) bool {
+	for i := range B4Size {
+		if h[i] <= b[i] {
+			return true
+		}
+	}
+	return false
+}
+
+// IsGreaterThanOrEqualTo returns true if h is greater than or equal to b.
+// This compares bytes from most significant to least significant in little-endian order.
+func (h B4) IsGreaterThanOrEqualTo(b B4) bool {
```

Comment:
if we have `Equals` and `IsLessThanOrEqualTo` (even just `isLess`)\,w e don't need `IsGreaterThanOrEqualTo`